### PR TITLE
Fix schema upgrade up to 224

### PR DESCRIPTION
**Pull request by margrietpalm**
## Incorrect control measure map

* Schematisation: https://management.staging.3di.live/schematisations/596/6066?organisation=b08433fa47c1401eb9cbd4156034c679
* Problem: `control_measure_map.geom` wasn't set properly because of incomplete conditions

## v2_impervious_surface and v2_impervious_surface_map mismatch

* Schematisation: https://management.staging.3di.live/schematisations/576/6023?organisation=b08433fa47c1401eb9cbd4156034c679
* Problem: not all items in `v2_impervious_surface` have a match with `v2_impervious_surface_map`
```sql
SELECT vis.id, vis.code, vis.nr_of_inhabitants , vis.area, vis.dry_weather_flow   
FROM v2_impervious_surface vis  
LEFT JOIN v2_impervious_surface_map   
ON vis.id = v2_impervious_surface_map.impervious_surface_id  
WHERE v2_impervious_surface_map.impervious_surface_id IS NULL AND   
vis.nr_of_inhabitants > 0 AND vis.dry_weather_flow  > 0;  
16870|10_14.047-inw|3.0|0.0|120.0  
17219|10_14.576-inw|7.0|0.0|120.0  
17220|10_14.578-inw|3.0|0.0|120.0
```
* Solution: remove lines without match in map and warn

## Several issues listed in #79

### CODE REVIEW

File: threedi_schema/migrations/versions/0223_upgrade_db_inflow.py [hunk 1]
```diff
@@ -157,6 +162,18 @@ def copy_v2_data_to_dry_weather_flow(src_table: str):
...
+def handle_null_geoms(src_table, tgt_table):
+    conn = op.get_bind()
+    missing = conn.execute(sa.text(f"select id from {tgt_table} where geom IS NULL")).fetchall()
+    if len(missing) > 0:
+        msg = (f"Could not create {tgt_table}.geom because {src_table}.id is not "
+               f"present in {src_table}_map. The following id's will not be "
+               f"migrated to {tgt_table}:  {missing}")
+        warnings.warn(msg, NullGeomWarning)
+        op.execute(sa.text(f"DELETE FROM {tgt_table} WHERE geom IS NULL;"))
+        rows_left = conn.execute(sa.text(f"select COUNT(*) from {tgt_table}")).fetchone()[0]
```
Comment (elisalle) [inline threedi_schema/migrations/versions/0223_upgrade_db_inflow.py:174]: what do you do with `rows_left` afterwards?

File: threedi_schema/migrations/versions/0223_upgrade_db_inflow.py [hunk 2]
```diff
@@ -358,6 +389,11 @@ def populate_surface_and_dry_weather_flow():
     copy_v2_data_to_dry_weather_flow(src_table)
     copy_v2_data_to_surface_map(f"{src_table}_map")
     copy_v2_data_to_dry_weather_flow_map(f"{src_table}_map")
+
+    # Check if any NULL geoms are left and remove them
+    # handle_null_geoms(src_table, 'surface')
+    # handle_null_geoms(src_table, 'dry_weather_flow')
```
Comment (elisalle) [inline threedi_schema/migrations/versions/0223_upgrade_db_inflow.py:395]: should these be commented?

File: threedi_schema/migrations/versions/0224_upgrade_db_structure_control.py [hunk 3]
```diff
@@ -166,19 +166,29 @@ def set_geom_for_control_measure_map():
...
+        UPDATE
             control_measure_map
-        SET 
+        SET
             geom = (
                 SELECT 
                     MakeLine(tc.geom, cml.geom)
                 FROM 
-                    {control_table} AS tc,
-                    control_measure_location AS cml
```
Comment (elisalle) [inline threedi_schema/migrations/versions/0224_upgrade_db_structure_control.py:177]: Does the rest of the query still work now that you no longer have defined cml? It is still referred to.

Comment (margrietpalm) [reply threedi_schema/migrations/versions/0224_upgrade_db_structure_control.py:177]: Yes, works

File: threedi_schema/migrations/versions/0223_upgrade_db_inflow.py (hunk 2 above)
Comment (margrietpalm) [reply threedi_schema/migrations/versions/0223_upgrade_db_inflow.py]: No, changed in cee0a5b4a671a731d6812d0ddef8b8bf862867d0

File: threedi_schema/migrations/versions/0223_upgrade_db_inflow.py (hunk 1 above)
Comment (margrietpalm) [reply threedi_schema/migrations/versions/0223_upgrade_db_inflow.py]: Nothing, removed in cee0a5b4a671a731d6812d0ddef8b8bf862867d0

**Review by elisalle — APPROVED** [review]

**Pull request merged**